sdk/python/kfp/components/types/custom_artifact_types_test.py

```diff
@@ -0,0 +1,595 @@
+# limitations under the License.
+
+import inspect
+import os
+import sys
+import tempfile
+import textwrap
+import typing
+from typing import Any
+import unittest
+
+from absl.testing import parameterized
+import kfp
+from kfp import compiler
+from kfp import dsl
+from kfp.components.types import artifact_types
+from kfp.components.types import custom_artifact_types
+from kfp.components.types import type_annotations
+from kfp.components.types.artifact_types import Artifact
+from kfp.components.types.artifact_types import Dataset
+from kfp.components.types.type_annotations import Input
+from kfp.components.types.type_annotations import InputPath
+from kfp.components.types.type_annotations import Output
+from kfp.components.types.type_annotations import OutputPath
+import typing_extensions
+import yaml
+
+Alias = Artifact
+artifact_types_alias = artifact_types
+
+
+class TestFuncToRootAnnotationSymbol(unittest.TestCase):
+
+    def test_no_annotations(self):
+
+        def func(a, b):
+            pass
+
+        actual = custom_artifact_types.func_to_root_annotation_symbols_for_artifacts(
+            func)
+        expected = {}
+        self.assertEqual(actual, expected)
+
+    def test_no_return(self):
+
+        def func(a: int, b: Input[Artifact]):
+            pass
+
+        actual = custom_artifact_types.func_to_root_annotation_symbols_for_artifacts(
+            func)
+        expected = {'b': 'Artifact'}
+        self.assertEqual(actual, expected)
+
+    def test_with_return(self):
+
+        def func(a: int, b: Input[Artifact]) -> int:
+            return 1
+
+        actual = custom_artifact_types.func_to_root_annotation_symbols_for_artifacts(
+            func)
+        expected = {'b': 'Artifact'}
+        self.assertEqual(actual, expected)
+
+    def test_multiline(self):
+
+        def func(
+            a: int,
+            b: Input[Artifact],
+        ) -> int:
+            return 1
+
+        actual = custom_artifact_types.func_to_root_annotation_symbols_for_artifacts(
+            func)
+        expected = {'b': 'Artifact'}
+        self.assertEqual(actual, expected)
+
+    def test_alias(self):
+
+        def func(a: int, b: Input[Alias]):
+            pass
+
+        actual = custom_artifact_types.func_to_root_annotation_symbols_for_artifacts(
+            func)
+        expected = {'b': 'Alias'}
+        self.assertEqual(actual, expected)
+
+    def test_long_form_annotation(self):
+
+        def func(a: int, b: Input[artifact_types.Artifact]):
+            pass
+
+        actual = custom_artifact_types.func_to_root_annotation_symbols_for_artifacts(
+            func)
+        expected = {'b': 'artifact_types'}
+        self.assertEqual(actual, expected)
+
+    def test_named_tuple(self):
+
+        def func(
+            a: int,
+            b: Input[Artifact],
+        ) -> typing.NamedTuple('MyNamedTuple', [('a', int), (
+                'b', Artifact), ('c', artifact_types.Artifact)]):
```

Comment:
Completely off the topic, but this remand me that we could have a workaround for using the same name of input and output with Python function component interface.

Reply:
Great catch. I didn't realize this when writing the test :)